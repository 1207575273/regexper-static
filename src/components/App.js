import React from 'react';

import Message from 'components/Message';

const App = () => (
  <Message heading="React App">
    <p>Placeholder app content</p>
  </Message>
);

export default App;
