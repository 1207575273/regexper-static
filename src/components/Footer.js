import React from 'react';

const Footer = () => (
  <footer>
    <ul className="inline with-separator">
      <li>
        Created by <a href="mailto:jeff.avallone@gmail.com">Jeff Avallone</a>
      </li>
      <li>
        Generated images licensed: <a rel="license" href="http://creativecommons.org/licenses/by/3.0/">
          <img
            alt="Creative Commons CC-BY-3.0 License"
            src="https://licensebuttons.net/l/by/3.0/80x15.png" />
        </a>
      </li>
    </ul>
  </footer>
);

export default Footer;
