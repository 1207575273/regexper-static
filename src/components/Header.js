import React from 'react';
import { translate, Trans } from 'react-i18next';

import GithubIcon from 'feather-icons/dist/icons/github.svg';

const Header = () => (
  <header id="main" data-banner={ process.env.BANNER }>
    <h1>
      <a href="/">Regexper</a>
    </h1>

    <ul className="inline">
      <li><a href="https://github.com/javallone/regexper-static">
        <GithubIcon/><Trans>Source on GitHub</Trans>
      </a></li>
    </ul>
  </header>
);

export default translate()(Header);
export { Header };
