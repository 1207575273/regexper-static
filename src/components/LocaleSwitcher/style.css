.switcher {
  margin-left: 1ex;
}
