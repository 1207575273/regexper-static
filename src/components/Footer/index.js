import React from 'react';
import { translate, Trans } from 'react-i18next';

import style from './style.css';

const Footer = () => (
  <footer>
    <ul>
      <li>
        <Trans>Created by <a href="mailto:jeff.avallone@gmail.com">Jeff Avallone</a></Trans>
      </li>
      <li>
        <Trans i18nKey="Generated images licensed">
          Generated images licensed: <a rel="license external noopener noreferrer" target="_blank" href="http://creativecommons.org/licenses/by/3.0/">
            <img
              alt="Creative Commons CC-BY-3.0 License"
              src="https://licensebuttons.net/l/by/3.0/80x15.png" />
          </a>
        </Trans>
      </li>
      <li>
        <a href="/privacy.html"><Trans>Privacy Policy</Trans></a>
      </li>
    </ul>
    <div className={ style.buildId }>{ process.env.BUILD_ID }</div>
  </footer>
);

export default translate()(Footer);
export { Footer };
