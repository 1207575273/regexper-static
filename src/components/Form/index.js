import React from 'react';
import PropTypes from 'prop-types';
import { translate, Trans } from 'react-i18next';

import DownloadIcon from 'feather-icons/dist/icons/download.svg';
import LinkIcon from 'feather-icons/dist/icons/link.svg';
import ExpandIcon from 'feather-icons/dist/icons/chevrons-down.svg';

import style from './style.css';

class Form extends React.Component {
  handleSubmit = event => {
    event.preventDefault();
    this.props.onSubmit.call(this, {
      expr: this.textarea.value,
      syntax: this.syntax.value
    });
  }

  handleKeyPress = event => {
    if (event.charCode === 13 && event.shiftKey) {
      this.handleSubmit(event);
    }
  }

  textareaRef = textarea => this.textarea = textarea

  syntaxRef = syntax => this.syntax = syntax

  permalinkAction() {
    const { permalinkUrl } = this.props;

    if (!permalinkUrl) {
      return;
    }

    return <li>
      <a href={ permalinkUrl }><LinkIcon/><Trans>Permalink</Trans></a>
    </li>;
  }

  downloadActions() {
    const { downloadUrls } = this.props;

    if (!downloadUrls) {
      return;
    }

    return downloadUrls.map(({ url, filename, type, label }, i) => <li key={ i }>
      <a href={ url } download={ filename } type={ type }>
        <DownloadIcon/>{ label }
      </a>
    </li>);
  }

  render() {
    const { syntaxes, t } = this.props;

    return <div className={ style.form }>
      <form onSubmit={ this.handleSubmit }>
        <textarea
          ref={ this.textareaRef }
          onKeyPress={ this.handleKeyPress }
          autoFocus
          placeholder={ t('Enter regular expression to display') }></textarea>
        <button type="submit"><Trans>Display</Trans></button>
        <div className={ style.select }>
          <select ref={ this.syntaxRef }>
            { Object.keys(syntaxes).map(id => (
              <option value={ id } key={ id }>{ syntaxes[id] }</option>
            )) }
          </select>
          <ExpandIcon/>
        </div>
        <ul className={ ['inline', 'with-separator', style.actions].join(' ') }>
          { this.downloadActions() }
          { this.permalinkAction() }
        </ul>
      </form>

    </div>;
  }
}

Form.propTypes = {
  syntaxes: PropTypes.object,
  onSubmit: PropTypes.func,
  permalinkUrl: PropTypes.string,
  downloadUrls: PropTypes.array,
  t: PropTypes.func
};

export default translate()(Form);
export { Form };
