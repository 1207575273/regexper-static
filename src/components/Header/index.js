import React from 'react';
import { translate, Trans } from 'react-i18next';

import style from './style.css';
import GithubIcon from 'feather-icons/dist/icons/github.svg';

import LocaleSwitcher from 'components/LocaleSwitcher';

const Header = () => (
  <header className={ style.header } data-banner={ process.env.BANNER }>
    <h1>
      <a href="/">Regexper</a>
    </h1>

    <ul>
      <li><a href="https://github.com/javallone/regexper-static" rel="external noopener noreferrer" target="_blank">
        <GithubIcon/><Trans>Source on GitHub</Trans>
      </a></li>
      <li>
        <a href="/privacy.html"><Trans>Privacy Policy</Trans></a>
      </li>
      <li><LocaleSwitcher /></li>
    </ul>
  </header>
);

export default translate()(Header);
export { Header };
