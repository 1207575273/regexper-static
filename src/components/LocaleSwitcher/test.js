jest.mock('components/SVG');

import React from 'react';
import { shallow } from 'enzyme';

import { LocaleSwitcher } from 'components/LocaleSwitcher';
import { translate } from '__mocks__/i18n';

describe('LocaleSwitcher', () => {
  test('rendering', () => {
    const component = shallow(
      <LocaleSwitcher t={ translate }/>
    );
    expect(component).toMatchSnapshot();
  });
});
