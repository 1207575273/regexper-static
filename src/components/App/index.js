import React from 'react';
import PropTypes from 'prop-types';
import { translate } from 'react-i18next';
import URLSearchParams from 'url-search-params';

import style from './style.css';

import Form from 'components/Form';
import Message from 'components/Message';
import SVG from 'components/SVG';
import { demoImage } from 'devel';

const syntaxes = {
  js: 'JavaScript',
  pcre: 'PCRE'
};

class App extends React.PureComponent {
  state = {}

  image = React.createRef()

  componentDidMount() {
    window.addEventListener('hashchange', this.handleHashChange);
    this.handleHashChange();
  }

  componentWillUnmount() {
    window.removeEventListener('hashchange', this.handleHashChange);
  }

  async setSvgUrl() {
    try {
      const type = 'image/svg+xml';
      const blob = await this.image.current.svgUrl(type);

      this.setState({
        svgUrl: {
          url: URL.createObjectURL(blob),
          label: this.props.t('Download SVG'),
          filename: 'image.svg',
          type
        }
      });
    }
    catch (e) {
      console.error(e); // eslint-disable-line no-console
    }
  }

  async setPngUrl() {
    try {
      const type = 'image/png';
      const blob = await this.image.current.pngUrl(type);

      this.setState({
        pngUrl: {
          url: URL.createObjectURL(blob),
          label: this.props.t('Download PNG'),
          filename: 'image.png',
          type
        }
      });
    }
    catch (e) {
      console.error(e); // eslint-disable-line no-console
    }
  }

  handleSubmit = ({expr, syntax}) => {
    if (expr) {
      const params = new URLSearchParams({ syntax, expr });
      document.location.hash = params.toString();
    }
  }

  handleHashChange = () => {
    const query = document.location.hash.slice(1);
    const params = new URLSearchParams(query);
    const { expr, syntax } = (() => {
      if (params.get('syntax')) {
        return {
          syntax: params.get('syntax'),
          expr: params.get('expr')
        };
      } else {
        // Assuming old-style URL
        return {
          syntax: 'js',
          expr: query
        };
      }
    })();

    if (!expr) {
      return;
    }

    console.log(syntax, expr); // eslint-disable-line no-console
    this.setState({
      image: demoImage,
      permalinkUrl: document.location.toString(),
      syntax,
      expr
    }, async () => {
      await this.image.current.doReflow();
      this.setSvgUrl();
      this.setPngUrl();
    });
  }

  render() {
    const { svgUrl, pngUrl, permalinkUrl, syntax, expr, image } = this.state;
    const downloadUrls = [
      svgUrl,
      pngUrl
    ].filter(Boolean);

    return <React.Fragment>
      <Form
        syntaxes={ syntaxes }
        downloadUrls={ downloadUrls }
        permalinkUrl={ permalinkUrl }
        syntax={ syntax }
        expr={ expr }
        onSubmit={ this.handleSubmit }/>
      <Message type="error" heading="Sample Error">
        <p>Sample error message</p>
      </Message>
      <Message type="warning" heading="Sample Warning">
        <p>Sample warning message</p>
      </Message>
      { image && <div className={ style.render }>
        <SVG data={ image } ref={ this.image }/>
      </div> }
    </React.Fragment>;
  }
}

App.propTypes = {
  t: PropTypes.func
};

export default translate()(App);
export { App };
