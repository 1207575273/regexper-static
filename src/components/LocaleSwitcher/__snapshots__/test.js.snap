// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocaleSwitcher rendering 1`] = `
<label>
  <Trans>
    Language
  </Trans>
  <select
    className="switcher"
    onChange={[Function]}
    value="en"
  />
</label>
`;
